src/xpi.js

```diff
@@ -102,4 +102,43 @@ export default class Xpi {
         .catch((err) => reject(err));
     });
   }
+
+  getFileAsString(path) {
+    return new Promise((resolve, reject) => {
+      return this.getFileAsStream(path)
+        .then((fileStream) => {
+          var fileString = '';
+          fileStream.on('data', (chunk) => {
+            fileString += chunk;
+          });
+
+          // Once the file is assembled, resolve the promise.
+          fileStream.on('end', () => {
+            resolve(fileString);
+          });
+        })
+        .catch((err) => reject(err));
+    });
+  }
+
+  getJSFiles() {
+    return new Promise((resolve, reject) => {
+      return this.getMetaData()
+        .then((metadata) => {
+          let jsFiles = [];
+
+          for (let filename in metadata) {
+            // TODO: Check for JS files better than this (follow require path)
```

Comment:
Good question - I wonder how the current validator deals with the possibility of imports including requires/imports of files that don't end in .js?